java/src/lib.rs

```diff
@@ -139,6 +141,60 @@ pub extern "system" fn Java_glide_ffi_resolvers_RedisValueResolver_valueFromPoin
     .unwrap_or(JObject::null())
 }
 
+/// Creates a leaked vector of byte arrays representing the args and returns a handle to it.
+///
+/// This function is meant to be invoked by Java using JNI.
+///
+/// * `env`     - The JNI environment.
+/// * `_class`  - The class object. Not used.
+/// * `args`    - The arguments. This should be a byte[][] from Java.
+#[no_mangle]
+pub extern "system" fn Java_glide_ffi_resolvers_RedisValueResolver_createLeakedBytesVec<'local>(
+    mut env: JNIEnv<'local>,
+    _class: JClass<'local>,
+    args: JObjectArray<'local>,
+) -> jlong {
+    handle_panics(
+        move || {
+            fn create_leaked_bytes_vec<'a>(
+                env: &mut JNIEnv<'a>,
+                args: JObjectArray<'a>,
+            ) -> Result<jlong, FFIError> {
+                let num_elements = env.get_array_length(&args)?;
+                let mut bytes_vec = Vec::with_capacity(num_elements as usize);
+
+                for index in 0..num_elements {
+                    let value = env.get_object_array_element(&args, index as jsize)?;
+                    bytes_vec.push(Bytes::from(
+                        env.convert_byte_array(JByteArray::from(value))?,
+                    ))
+                }
+                Ok(Box::leak(Box::new(bytes_vec)) as *mut Vec<Bytes> as jlong)
+            }
+            let result = create_leaked_bytes_vec(&mut env, args);
+            handle_errors(&mut env, result)
+        },
+        "createLeakedBytesVec",
+    )
+    .unwrap_or(0)
+}
+
+/// Returns the maximum total length of request arguments.
+///
+/// This function is meant to be invoked by Java using JNI.
+///
+/// * `_env`    - The JNI environment. Not used.
+/// * `_class`  - The class object. Not used.
+#[no_mangle]
+pub extern "system" fn Java_glide_ffi_resolvers_RedisValueResolver_getMaxRequestArgsLength<
+    'local,
+>(
+    mut _env: JNIEnv<'local>,
```

Comment:
I don't know if the mut prefix is necessary here. I don't know if it affects a signature lookup.

Reply:
in rust unused variables are prefixed with `_`

Reply:
Yeah the comment more was on the signature resolution (in C++ marking const in front of parameter is significant to its signature).